google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -3547,4 +3547,36 @@ HmacKeyMetadata updateHmacKeyState(
    * @throws StorageException upon failure
    */
   ServiceAccount getServiceAccount(String projectId);
+
+  /**
+   * Creates a notification with the specified entity on the specified bucket.
+   *
+   * @return the notification that was created.
+   * @throws StorageException upon failure
+   */
+  Notification createNotification(String bucket, NotificationInfo notificationInfo);
+
+  /**
+   * Get the notification with the specified name on the specified object.
+   *
+   * @return the notification object that exist on the bucket.
+   * @throws StorageException upon failure
+   */
+  Notification getNotification(String bucket, String notification);
+
+  /**
+   * List the notifications for the provided bucket.
+   *
+   * @return a list of {@link Notification} objects that exist on the bucket.
+   * @throws StorageException upon failure
+   */
+  List<Notification> listNotifications(String bucket);
+
+  /**
+   * Deletes the notification with the specified name on the specified object.
+   *
+   * @return {@code true} if the notification was deleted, {@code false} if it was not found
+   * @throws StorageException upon failure
```

Comment:
Does `delete` mean unsubscribe?

the specified object -> the bucket
was deleted -> has been deleted
if it was not found -> if not found